invenio_sip2/actions/actions.py

```diff
@@ -272,26 +194,137 @@ def execute(self, message, client):
+
+    @check_selfcheck_authentication
+    def execute(self, message, **kwargs):
+        """Execute action."""
+        institution_id = message.get_field_value('institution_id')
+        patron_id = message.get_field_value('patron_id')
+        selfcheck_language = message.i18n_language
+
+
 class EndPatronSession(Action):
     """Action to end patron session on automated circulation system."""
 
     @check_selfcheck_authentication
     def execute(self, message, client):
         """Execute action."""
-        # prepare message based on fixed fields
+        # prepare message based on required fields
         response_message = self.prepare_message_response(
-            end_session='Y',
+            end_session=True,
             transaction_date=acs_system.sip2_current_date,
+            institution_id=client.institution_id,
+            patron_id=message.get_field_value('patron_id')
         )
 
-        # add variable field
-        response_message.add_variable_field(
-            field_name='institution_id',
-            field_value=client.institution_id
+        # TODO: add optional fields
+        client.clear_patron_session()
+        return str(response_message)
+
+
+class ItemInformation(Action):
+    """Action to get item information from automated circulation system."""
+
+    @check_selfcheck_authentication
+    def execute(self, message, client):
+        """Execute action."""
+        patron_session = client.get_current_patron_session()
+        item_identifier = message.get_field_value('item_id')
+        item_information = item_handler(
+            client.remote_app, patron_session.get('patron_id'),
+            item_identifier, language=patron_session.get('language')
         )
-        response_message.add_variable_field(
-            field_name='patron_id',
-            field_value=message.get_field_value('patron_id')
+
+        # prepare message based on required fields
+        response_message = self.prepare_message_response(
+            circulation_status=get_circulation_status(
+                item_information.get('circulation_status')),
+            security_marker=get_security_marker_type(
+                item_information.get('security_marker')),
+            fee_type=item_information.get('fee_type'),
+            transaction_date=acs_system.sip2_current_date,
+            item_id=item_information.get('item_id'),
+            title_id=item_information.get('title_id')
         )
 
+        # add optional fields
+        for optional_field in self.optional_fields:
+            response_message.add_field(
+                field=optional_field,
+                field_value=item_information.get(optional_field.name)
+            )
+
         return str(response_message)
+
+
+class BlockPatron(Action):
+    """Action to block patron."""
+
+    @check_selfcheck_authentication
+    def execute(self, message, client, **kwargs):
+        """Execute action."""
```

Comment:
in next PR : maybe add a TODO comment to retrieve easily these 'dummy' methods

Reply:
Ok. Done